server/server.js

```diff
@@ -29,14 +30,38 @@ var server = app.listen(process.env.PORT || 3000, function(){
   console.log('listening to port: ' + 3000);
 });
 
+app.use(passport.initialize());
+app.use(passport.session());
+
 //ROUTES-------------------------
-app.get('/repos/:repoOwner/:repoName/commits', commitsController.getCommits);
+app.get('/getAccessToken',
+  passport.authenticate('github', { failureRedirect: '/pageNotFound' }),
+  function(req, res){
+    repoFullName = req.query.repoFullName;
+    console.log(repoFullName);
+    res.redirect('/');
+  }
+);
+
+app.get('/repos/:repoOwner/:repoName/commits',
```

Comment:
this changes the behavior to always require login before checking if the commits are already in the db